src/budget_monitor.py

```
@@ -0,0 +1,832 @@
+            return
+
+        self.cost_tracker = cost_tracker
+        self.budget_limit = budget_limit
+        self._monitoring_active = True
+
+        self._monitoring_thread = threading.Thread(
+            target=self._monitoring_loop, daemon=True
+        )
+        self._monitoring_thread.start()
+
+        print(f"👁️ Monitoramento de budget iniciado")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)